lib/Room.php

```diff
@@ -729,6 +755,42 @@ public function setReadOnly(int $newState): bool {
 		return true;
 	}
 
+	/**
+	 * @param int $newState New listable scope from self::LISTABLE_*
+	 * 						Also it's only allowed on rooms of type
+	 * 						`self::GROUP_CALL` and `self::PUBLIC_CALL`
+	 * @return bool True when the change was valid, false otherwise
+	 */
+	public function setListable(int $newState): bool {
+		$oldState = $this->getListable();
+		if ($newState === $oldState) {
+			return true;
+		}
+
+		if (!in_array($this->getType(), [self::GROUP_CALL, self::PUBLIC_CALL, self::CHANGELOG_CONVERSATION], true)) {
+			return false;
+		}
+
+		if ($newState < 0 || $newState > 3) {
```

Comment:
`!in_array` woth the constants? Makes it future saver to extend by searching for the constants